### DISCUSSION TITLE
How to use the self-designed Simulink yaw control system to conduct co-simulation with OpenFAST

### DISCUSSION THREAD

**Discussion by yiyan919** [NatLabRockies/ROSCO#407] [Q&A]
Hello everyone, I plan to design a yaw control system based on fuzzy PID in simulink. I only know that I need to set YCmode to 4 in the servoDyn.dat file, but I don't know where to add my own system. By the way, I have only built the fuzzy PID control module at present. Do I need to build a complete yaw control system including motor, reducer, etc.
Sincerely hope to get your reply and help!

**Comment by dzalkind** [comment]
I suggest you review the inputs to the openfast model here: https://github.com/OpenFAST/openfast/blob/main/glue-codes/simulink/examples/OpenLoop.mdl

The inputs to the simulink model are yaw position and yaw rate, so you don't need to model the motor, etc. unless you want to do that yourself in simulink.

**Reply by yiyan919** [reply]
Thank you very much for your reply! Let me try it！
